src/lrf.rs

```diff
@@ -255,23 +252,21 @@ fn sgrproj_box_sum_fastx_r2<T: Pixel>(a: &mut u32, b: &mut u32,
                                       stripe_y: isize, stripe_h: usize,
                                       x: isize, y: isize,
                                       s: u32, bdm8: usize,
-                                      backing: &PlaneSlice<T>, backing_h: usize,
-                                      cdeffed: &PlaneSlice<T>, cdeffed_h: usize) {
+                                      backing: &PlaneSlice<T>, cdeffed: &PlaneSlice<T>,
+                                      crop_h: usize) {
   let mut ssq = 0;
   let mut sum = 0;
   for yi in y - 2..=y + 2 {
+    // clamp to crop rectangle before deciding source.
+    let clamp_y = clamp(yi, -cdeffed.y, crop_h as isize - 1);
```

Comment:
Maybe this logic could be factored out since it is used in two places (and then we could put a note on it to remember why it exists).